FreeIPA: on a replica, ipa ca-del subca fails with @pki/master repo

The following issue happens when installing pki from @pki/master repository.
Scenario: install a server + replica with CA service, create a subca on the server, disable the subca on the replica and delete the subca on the replica. The command `ipa ca-del` fails with `ipa: ERROR: Request failed with status 500: Non-2xx response from CA REST API: 500. `

Reproducer:
On the server:
```
dnf copr enable -y @pki/master
dnf copr enable -y @freeipa/freeipa-master-nightly
dnf install -y freeipa-server-dns
ipa-server-install --domain ipa.test --realm IPA.TEST --setup-dns --auto-forwarders -a Secret123 -p Secret123 -U
```
On the replica:
```
dnf copr enable -y @pki/master
dnf copr enable -y @freeipa/freeipa-master-nightly
dnf install -y freeipa-server-dns
ipa-replica-install --domain ipa.test --realm IPA.TEST --setup-dns --auto-forwarders --setup-ca --principal admin --password Secret123 --server server.ipa.test -U
```

Create a sub ca on the server:
```
echo Secret123 | kinit admin
ipa ca-add subca --subject cn=subca,O=IPA.TEST
```

Disable and delete the sub ca on the replica:
```
echo Secret123 | kinit admin
ipa ca-disable subca
ipa ca-del subca
```

The last command fails.

The replica logs show that PKI is trying to find a subca with a dn cn=null,ou=certificateRepository,ou=ca,o=ipaca:
```
2024-02-15 07:47:06 [ajp-nio-0:0:0:0:0:0:0:1-8009-exec-5] INFO: AuthorityService: deleting authority 826c3884-348f-4fde-86aa-5aa07c8ed878
2024-02-15 07:47:06 [ajp-nio-0:0:0:0:0:0:0:1-8009-exec-5] INFO: LDAPSession: Retrieving cn=null,ou=certificateRepository, ou=ca,o=ipaca
2024-02-15 07:47:06 [ajp-nio-0:0:0:0:0:0:0:1-8009-exec-5] SEVERE: Error modifying authority: Record not found
Record not found
        at com.netscape.cmscore.dbs.LDAPSession.read(LDAPSession.java:200)
        at com.netscape.cmscore.dbs.LDAPSession.read(LDAPSession.java:149)
        at com.netscape.cmscore.dbs.CertificateRepository.readCertificateRecord(CertificateRepository.java:819)
        at org.dogtagpki.server.ca.CAEngine.revokeAuthority(CAEngine.java:2184)
        at org.dogtagpki.server.ca.CAEngine.deleteAuthority(CAEngine.java:2256)
        at org.dogtagpki.server.ca.rest.AuthorityService.deleteCA(AuthorityService.java:424)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base/java.lang.reflect.Method.invoke(Method.java:568)
        at org.jboss.resteasy.core.MethodInjectorImpl.invoke(MethodInjectorImpl.java:140)
        at org.jboss.resteasy.core.ResourceMethodInvoker.invokeOnTarget(ResourceMethodInvoker.java:295)
        at org.jboss.resteasy.core.ResourceMethodInvoker.invoke(ResourceMethodInvoker.java:249)
        at org.jboss.resteasy.core.ResourceMethodInvoker.invoke(ResourceMethodInvoker.java:236)
        at org.jboss.resteasy.core.SynchronousDispatcher.invoke(SynchronousDispatcher.java:406)
        at org.jboss.resteasy.core.SynchronousDispatcher.invoke(SynchronousDispatcher.java:213)
        at org.jboss.resteasy.plugins.server.servlet.ServletContainerDispatcher.service(ServletContainerDispatcher.java:228)
        at org.jboss.resteasy.plugins.server.servlet.HttpServletDispatcher.service(HttpServletDispatcher.java:56)
        at org.jboss.resteasy.plugins.server.servlet.HttpServletDispatcher.service(HttpServletDispatcher.java:51)
        at javax.servlet.http.HttpServlet.service(HttpServlet.java:623)
        at jdk.internal.reflect.GeneratedMethodAccessor44.invoke(Unknown Source)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base/java.lang.reflect.Method.invoke(Method.java:568)
        at org.apache.catalina.security.SecurityUtil.lambda$execute$0(SecurityUtil.java:222)
        at java.base/java.security.AccessController.doPrivileged(AccessController.java:712)
        at java.base/javax.security.auth.Subject.doAsPrivileged(Subject.java:584)
        at org.apache.catalina.security.SecurityUtil.execute(SecurityUtil.java:250)
        at org.apache.catalina.security.SecurityUtil.doAsPrivilege(SecurityUtil.java:142)
        at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:207)
        at org.apache.catalina.core.ApplicationFilterChain.lambda$doFilter$0(ApplicationFilterChain.java:137)
        at java.base/java.security.AccessController.doPrivileged(AccessController.java:569)
        at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:136)
        at org.apache.tomcat.websocket.server.WsFilter.doFilter(WsFilter.java:51)
        at jdk.internal.reflect.GeneratedMethodAccessor43.invoke(Unknown Source)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base/java.lang.reflect.Method.invoke(Method.java:568)
        at org.apache.catalina.security.SecurityUtil.lambda$execute$0(SecurityUtil.java:222)
        at java.base/java.security.AccessController.doPrivileged(AccessController.java:712)
        at java.base/javax.security.auth.Subject.doAsPrivileged(Subject.java:584)
        at org.apache.catalina.security.SecurityUtil.execute(SecurityUtil.java:250)
        at org.apache.catalina.security.SecurityUtil.doAsPrivilege(SecurityUtil.java:202)
        at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:176)
        at org.apache.catalina.core.ApplicationFilterChain.lambda$doFilter$0(ApplicationFilterChain.java:137)
        at java.base/java.security.AccessController.doPrivileged(AccessController.java:569)
        at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:136)
        at org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:168)
        at org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:90)
        at org.apache.catalina.authenticator.AuthenticatorBase.invoke(AuthenticatorBase.java:596)
        at com.netscape.cms.tomcat.ExternalAuthenticationValve.invoke(ExternalAuthenticationValve.java:83)
        at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:130)
        at org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:93)
        at org.apache.catalina.valves.rewrite.RewriteValve.invoke(RewriteValve.java:545)
        at org.apache.catalina.valves.AbstractAccessLogValve.invoke(AbstractAccessLogValve.java:670)
        at org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:74)
        at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:342)
        at org.apache.coyote.ajp.AjpProcessor.service(AjpProcessor.java:424)
        at org.apache.coyote.AbstractProcessorLight.process(AbstractProcessorLight.java:63)
        at org.apache.coyote.AbstractProtocol$ConnectionHandler.process(AbstractProtocol.java:928)
        at org.apache.tomcat.util.net.NioEndpoint$SocketProcessor.doRun(NioEndpoint.java:1794)
        at org.apache.tomcat.util.net.SocketProcessorBase.run(SocketProcessorBase.java:52)
        at org.apache.tomcat.util.threads.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1191)
        at org.apache.tomcat.util.threads.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:659)
        at org.apache.tomcat.util.threads.TaskThread$WrappingRunnable.run(TaskThread.java:61)
        at java.base/java.lang.Thread.run(Thread.java:840)
Caused by: netscape.ldap.LDAPException: No such object (32); matchedDN = ou=certificateRepository,ou=ca,o=ipaca
        at netscape.ldap.LDAPConnection.checkMsg(LDAPConnection.java:4933)
        at netscape.ldap.LDAPConnection.checkSearchMsg(LDAPConnection.java:2686)
        at netscape.ldap.LDAPConnection.search(LDAPConnection.java:2658)
        at netscape.ldap.LDAPConnection.search(LDAPConnection.java:2484)
        at com.netscape.cmscore.dbs.LDAPSession.read(LDAPSession.java:178)
        ... 63 more

2024-02-15 07:47:06 [ajp-nio-0:0:0:0:0:0:0:1-8009-exec-5] INFO: PKIExceptionMapper: Returning PKIException
2024-02-15 07:47:06 [ajp-nio-0:0:0:0:0:0:0:1-8009-exec-5] INFO: PKIExceptionMapper: XML exception:
<?xml version="1.0" encoding="UTF-8" standalone="no"?>
<PKIException>
    <ClassName>com.netscape.certsrv.base.PKIException</ClassName>
    <Attributes/>
    <Code>500</Code>
    <Message>Error modifying authority: Record not found</Message>
</PKIException>
```

Version: dogtag-pki-ca-11.5.0-0.4.alpha7.20240214145232UTC.a064ebe7.fc39.noarch

--- a/base/ca/src/main/java/com/netscape/ca/AuthorityMonitor.java
+++ b/base/ca/src/main/java/com/netscape/ca/AuthorityMonitor.java
@@ -378,28 +378,6 @@ public class AuthorityMonitor implements Runnable {
         nsUniqueIds.remove(aid);
     }
 
-    public synchronized void trackUpdate(AuthorityID aid, LDAPControl[] responseControls) {
-
-        LDAPPostReadControl control = (LDAPPostReadControl)
-            LDAPUtil.getControl(LDAPPostReadControl.class, responseControls);
-
-        LDAPEntry entry = control.getEntry();
-
-        LDAPAttribute attr = entry.getAttribute("entryUSN");
-        if (attr != null) {
-            BigInteger entryUSN = new BigInteger(attr.getStringValueArray()[0]);
-            logger.debug("AuthorityMonitor: tracking entryUSN: " + entryUSN);
-            entryUSNs.put(aid, entryUSN);
-        }
-
-        attr = entry.getAttribute("nsUniqueId");
-        if (attr != null) {
-            String nsUniqueId = attr.getStringValueArray()[0];
-            logger.info("AuthorityMonitor: tracking nsUniqueId: " + nsUniqueId);
-            nsUniqueIds.put(aid, nsUniqueId);
-        }
-    }
-
     /**
      * Stop the activityMonitor thread
      *

--- a/base/ca/src/main/java/org/dogtagpki/server/ca/CAEngine.java
+++ b/base/ca/src/main/java/org/dogtagpki/server/ca/CAEngine.java
@@ -1723,8 +1723,6 @@ public class CAEngine extends CMSEngine {
         } finally {
             connectionFactory.returnConn(conn);
         }
-
-        authorityMonitor.trackUpdate(authorityID, responseControls);
     }
 
     public synchronized void modifyAuthorityEntry(AuthorityID aid, LDAPModificationSet mods) throws EBaseException {
@@ -1743,8 +1741,6 @@ public class CAEngine extends CMSEngine {
         } finally {
             connectionFactory.returnConn(conn);
         }
-
-        authorityMonitor.trackUpdate(aid, responseControls);
     }
 
     public synchronized void deleteAuthorityEntry(AuthorityID aid) throws EBaseException {

